AutomationHelpers/src/Modules/CollectErrorMessagesModule.cs

```diff
@@ -0,0 +1,142 @@
+using Ranorex.Core.Testing;
+using Ranorex.Core.Reporting;
+
+namespace Ranorex.AutomationHelpers.Modules
+{
+    /// <summary>
+    /// Functionality: Collects all error messages from given test container.
+    /// Usage: Use module within a test container with enabled retry count
+    /// </summary>
+    [TestModule("E4E50BBC-20F8-461F-B39A-863B6F69DE89", ModuleType.UserCode, 1)]
+    public class CollectErrorMessagesModule : ITestModule
+    {
+        public static List<string> builder = new List<string>();
+
+        private static string lastTestContainerName = String.Empty;
+        private static int retryIteration;
+        private static bool registered = false;
+
+        public CollectErrorMessagesModule()
+        {
+            // Do not delete - a parameterless constructor is required!
+        }
+
+        void ITestModule.Run()
+        {
+            Mouse.DefaultMoveTime = 300;
+            Keyboard.DefaultKeyPressTime = 100;
+            Delay.SpeedFactor = 1.0;
+
+            //Delegate must be registered only once
+            if (!CollectErrorMessagesModule.registered)
+            {
+                //PDF will be generated at the very end of the TestSuite
+                TestSuite.TestSuiteCompleted += delegate
+                {
+                    LogErrorMessages();
+                };
+
+                CollectErrorMessagesModule.registered = true;
+            }
+
+            //Store name of TestContainer and reset retryIteration in case of new TestContainer
+            if (!String.IsNullOrEmpty(lastTestContainerName))
+            {
+                if (lastTestContainerName != TestReport.CurrentTestContainerActivity.FullDisplayName)
+                {
+                    CollectErrorMessagesModule.retryIteration = 1;
+                }
+                else
+                {
+                    CollectErrorMessagesModule.retryIteration++;
+                }
+            }
+
+            if (String.IsNullOrEmpty(lastTestContainerName))
+            {
+                CollectErrorMessagesModule.lastTestContainerName = TestReport.CurrentTestContainerActivity.FullDisplayName;
+                CollectErrorMessagesModule.retryIteration = 1;
+            }
+
+            //GetErrorMessages from TestContainer   
+            var container = TestReport.CurrentTestContainerActivity;
+
+            //Only collect error messages if the test run failed
+            if (container.Status == ActivityStatus.Failed)
+            {
+                GetErrorMessages((TestContainerActivity)container);
+            }
+        }
+
+        /// <summary>
+        /// Recursively collects all error messages from given test container
+        /// </summary>
+        /// <param name="container"></param>
+        private void GetErrorMessages(TestContainerActivity container)
+        {
+            foreach (var containerChild in container.Children)
+            {
+                var containerChildType = containerChild.GetType().Name;
```

Comment:
Uhhh, don't use type name (string), but the type directly, e.g.
if (containerChild is ITestContainerActivity)
{
// do something
}